store/src/java/com/zimbra/cs/account/ldap/LdapProvisioning.java

```diff
@@ -10721,4 +10721,22 @@ public boolean isCacheEnabled() {
         return useCache;
     }
 
+    @Override
+    public void refreshUserCredentials(Account account) throws ServiceException {
+        ZLdapContext zlc = null;
+
+        try {
+            zlc = LdapClient.getContext(LdapServerType.MASTER, LdapUsage.GET_ENTRY);
```

Comment:
It isn't clear we need to burden the master with this request.  If LDAP replication is working properly there should be very little delay between LDAP servers.  Generally speaking I don't think we're trying to force a conversation with the master unless we're doing writes.

Reply:
So you are saying I should swap the type of LdapServerType to what?
You are correct I don't care which LDAP server services the request I just know at this point that we likely have a stale password hash cached in the Provisioning instance (specifically in the imapd service in this case)

Reply:
It appears REPLICA is the only other option, but I get the feeling we shouldn't need to make that decision here. Perhaps we should just call a helper?  Maybe something that uses refreshEntry()/reload() or similar to avoid only getting a partially updated user entry (and then extending the lifetime of that potentially partially updated object).  Doing a partial update and then extending the cache time could lead to less desirable behavior (e.g. change in COS or more, that is expected to be updated because they occurred around the same time).